hpx/util/invoke.hpp

```diff
@@ -162,24 +162,20 @@ namespace hpx { namespace util
     HPX_CONSTEXPR HPX_HOST_DEVICE
     R invoke_r(F&& f, Ts&&... vs)
     {
-        return detail::invoke_impl<R,typename std::decay<F>::type>()(
-            std::forward<F>(f), std::forward<Ts>(vs)...);
+        return HPX_INVOKE_R(R, std::forward<F>(f), std::forward<Ts>(vs)...);
     }
 
     ///////////////////////////////////////////////////////////////////////////
-    /// \cond NOINTERNAL
     namespace functional
     {
         struct invoke
         {
             template <typename F, typename... Ts>
             HPX_CONSTEXPR HPX_HOST_DEVICE
             typename util::invoke_result<F, Ts...>::type
-            operator()(F && f, Ts &&... vs) const
+            operator()(F && f, Ts &&... vs)
             {
-                typedef typename util::invoke_result<F, Ts...>::type R;
-
-                return hpx::util::void_guard<R>(), util::invoke(std::forward<F>(f),
+                return util::invoke(std::forward<F>(f),
```

Comment:
Couldn't this use `HPX_INVOKE` instead (saving another call level)?

Reply:
Yes, they could be. These did not show up anywhere, they seem to be unused.